### ISSUE TITLE
Deterministic Dust Limit Computation

### ISSUE THREAD

**Issue by nkohen** [discreetlogcontracts/dlcspecs#11]
In [transaction construction](https://github.com/discreetlogcontracts/dlcspecs/blob/master/Transactions.md), both parties must always agree on which outputs end up on each transaction in order to make signing possible.

This requires we define a deterministic function `belowDustLimit` that takes in an output and the `feerate` at time of the DLC offer message, and returns true or false. If it returns true, both parties should remove that output from the transaction in question.

I have opened this issue to allow there to be a place for the discussion of what this function should look like to happen.

Some useful links:
- https://bitcoin.stackexchange.com/a/41082
- https://github.com/bitcoin/bitcoin/commit/9022aa3

**Comment by Tibo-lg** [comment]
I personally liked the motivation in Murch's thesis to discard output that are less than the cost of having the output (which I think it a bit different than the definition of dust).
It's in [here](http://murch.one/wp-content/uploads/2016/11/erhardt2016coinselection.pdf) section 3.5.5

Copy paste:
```
When encountering a tiny remainder after coin selection, it may be economically
beneficent to add it to the transaction fee instead of creating a change output. In
the case that the discarded amount were less than the cost of creating a change
output, the user actually would save costs by paying more fees. Therefore, the
transaction becomes more attractive for miners to select due to the higher fee.
```

**Comment by nkohen** [comment]
I agree that we definitely want to get rid of outputs that aren't economically viable. The main question then is what fee rate to use (for now probably makes the most sense to just use the fee rate at contract construction for simplicity) and also we will want to make sure we don't have any outputs (even economically viable ones) which violate relay policy.

**Comment by nkohen** [comment]
During our meeting, we agreed that we should specify removal of outputs that are not economically viable (cost more to spend than they are worth), as well as those beneath relay policy thresholds.

**Comment by Tibo-lg** [comment]
Just an update on this. I decided to go with a slightly modified version of the [Bitcoin core implementation of GetDustThreshold](https://github.com/bitcoin/bitcoin/blob/c536dfbcb00fb15963bf5d507b7017c241718bf6/src/policy/policy.cpp#L18) to determine if an output should be ignored or not based on a given fee rate. I guess that could be a good default?

**Comment by nkohen** [comment]
The plan is still to specify this dust computation for economic viability at some point but for now just to get the spec moving forward, I've just specified a `1000 satoshi` dust limit in #51

**Comment by ghost** [comment]
**TLDR:** We should use default value used by Bitcoin Core for `dustRelayFee`

1. DUST is confusing at least how it works in Bitcoin Core

2. Not sure if an output that was considered DUST last year will be the same right now or in next few months considering change in BTCUSD.

3. https://github.com/bitcoin/bitcoin/blob/3caee16946575e71e90ead9ac531f5a3a1259307/src/policy/policy.cpp#L16 mentioned the unit as sat/kB, almost everything else is mentioned in BTC/kB in Bitcoin Core which maybe changed to sat/vB in future. Few PRs and related discussion in this thread: https://twitter.com/jonatack/status/1368737836581617666

4. Few people have been trying to experiment with fee rate below 1 sat/vB although not successful: https://bitcoin.stackexchange.com/questions/99089/can-we-use-fee-rate-less-than-1-sat-vbyte

5. I tried my best to get some response about changing minimum default fee rate below 1 sat/vB, there was one reply from Pieter Wuille and doesn't look it will be changed:
https://bitcoin.stackexchange.com/questions/100861/why-cant-default-minimum-fee-rate-be-changed-to-0-1-0-2-sat-vbyte
